django_walletpass/classviews.py

```diff
@@ -37,12 +40,19 @@ def list(self, request, device_library_id, pass_type_id):
             return Response({}, status=status.HTTP_400_BAD_REQUEST)
 
         if 'passesUpdatedSince' in request.GET:
-            passes = passes.filter(updated_at__gt=datetime.strptime(request.GET['passesUpdatedSince'], FORMAT))
+            try:
+                # must be able to read UTC isoformat with TZ and FORMAT datetime string
+                # as well
+                dt = parse(request.GET['passesUpdatedSince'])
+                passes = passes.filter(updated_at__gt=dt)
+            except NonExistentTimeError:
+                dt = dt.replace(hour=0, minute=0)
```

Comment:
wouldn't `dt` be `None` here?

```
dt = dt.replace(hour=0.....
 ----^^----
```

Reply:
https://docs.python.org/3/library/datetime.html#datetime.date.replace don't think so.

Reply:
He is asking because `dt` is inside the `try` and he thinks that `parse()` is throwing the exception. The exception is actually being thrown by `filter()`, so `dt` cannot be None on line 49. Putting `dt` inside the `try` is a bit weird, but perfectly valid.

Reply:
```
In [4]: parse('2013-03-31 02:30:00')
Out[4]: datetime.datetime(2013, 3, 31, 2, 30)

In [5]: Pass.objects.filter(updated_at__gte=parse('2013-03-31 02:30:00'))
/usr/local/lib/python3.8/site-packages/django/db/models/fields/__init__.py:1416: RuntimeWarning: DateTimeField Pass.updated_at received a naive datetime (2013-03-31 02:30:00) while time zone support is active.
  warnings.warn("DateTimeField %s received a naive datetime (%s)"
---------------------------------------------------------------------------
NonExistentTimeError                      Traceback (most recent call last)
```

the error happens in the ORM, hence `dt` should be there. Am I right?

Reply:
@NachE you were faster :)

Reply:
Oh, okey. Good :)